### PR TITLE
Show correct number in series truncation.

### PR DESCRIPTION
**Pull request by thomasstep**
Change math in collection to represent the correct number of hidden posts. I might need some help with writing tests if necessary.

## What type of PR is this? (check all applicable)

- [ ] Refactor
- [ ] Feature
- [x] Bug Fix
- [ ] Optimization
- [ ] Documentation Update

## Description
This fixes https://github.com/forem/forem/issues/10956. Since the posts past the first two and before the last two are truncated, the math was slightly off. Instead of accounting for all four visible posts it only accounting for two.

## Related Tickets & Documents
Closes issue #10956 

## QA Instructions, Screenshots, Recordings

I tried to recreate and test locally before pushing, but my local stack would not let me publish posts.

### UI accessibility concerns?

This should only change a single text element. There should not be any accessibility concerns.

## Added tests?

- [ ] Yes
- [ ] No, and this is why: _please replace this line with details on why tests
      have not been included_
- [x] I need help with writing tests

## [Forem core team only] How will this change be communicated?

_Will this PR introduce a change that impacts Forem members or creators, the
development process, or any of our internal teams? If so, please note how you
will share this change with the people who need to know about it._

- [ ] I've updated the [Developer Docs](https://docs.forem.com) and/or
      [Admin Guide](https://forem.gitbook.io/forem-admin-guide/), or
      [Storybook](https://storybook.forem.com/) (for Crayons components)
- [ ] I've updated the README or added inline documentation
- [ ] I will share this change in a [Changelog](https://forem.dev/t/changelog)
      or in a [forem.dev](http://forem.dev) post
- [ ] I will share this change internally with the appropriate teams
- [ ] I'm not sure how best to communicate this change and need help
- [x] This change does not need to be communicated, and this is why not: this bug fix does not impact Forem members or creators, the development process, or our internal teams in such a way that it warrants documentation updates, communication to internal teams, etc. It will, however, be communicated that the issue has been resolved on the issue itself.

## [optional] What gif best describes this PR or how it makes you feel?

Farmer meme: It ain't much but it's honest work

### CODE REVIEW

**Comment by juliannatetreault** [comment]
Thank you so much for your contribution, @thomasstep — your work is greatly appreciated!

In the future, though, if you come across an issue that has already been assigned to someone, would you please first ask the assignee if they’re still interested in working on the issue and communicate with them that if not, you’re interested in taking on the issue and implementing a solution prior to opening a PR for the issue? This ensures that everyone has a chance to try their hand at a solution if they’re interested in doing so! 😊

**Comment by thomasstep** [comment]
>In the future, though, if you come across an issue that has already been assigned to someone, would you please first ask the assignee if they’re still interested in working on the issue and communicate with them that if not, you’re interested in taking on the issue and implementing a solution prior to opening a PR for the issue?

@juliannatetreault Yes ma'am I will. I apologize about that.

**Comment by juliannatetreault** [comment]
> @juliannatetreault Yes ma'am I will. I apologize about that.

Thanks so much and no worries at all, @thomasstep! 😄

**Review by juliannatetreault — COMMENTED** [review]
Thanks so much for taking the time to fix this bug, @thomasstep!

I tested your work out locally and it works just as expected! That said, I think that it would be beneficial if we added a test for your fix to `spec/requests/collections_spec.rb` to ensure that we catch any bugs in the future that may pop up. Let me know if you have any questions or need any help, otherwise, excellent fix!

**Comment by thomasstep** [comment]
@juliannatetreault That's good to hear! I've been messing with the spec that you referenced and I can't seem to get a test working to validate the text written by the change. Maybe I don't completely know what I changed, but I poked around and was thinking maybe writing a new spec in the `views/` folder. Am I totally off base there? It's probably also worth mentioning that I haven't ever written tests with `rspec`.

**Comment by juliannatetreault** [comment]
> @juliannatetreault That's good to hear! I've been messing with the spec that you referenced and I can't seem to get a test working to validate the text written by the change. Maybe I don't completely know what I changed, but I poked around and was thinking maybe writing a new spec in the `views/` folder. Am I totally off base there? It's probably also worth mentioning that I haven't ever written tests with `rspec`.

No worries, @thomasstep! Here's what I was thinking we could do to get the spec set up properly: 

- Within `spec/factories/collections.rb`, we could update the `:with_articles` trait to accept an `amount`. Here's an example of what we're looking to from the `factory_bot` documentation: https://github.com/thoughtbot/factory_bot/blob/master/GETTING_STARTED.md#with-transient-attributes

By doing this, we will be able to pass in a custom, larger amount of `articles` to our `collection` in the spec (`spec/requests/collections_spec.rb`) so that we can test that we get the expected `collection.articles.count` back and with that, the correct text that we expect to see in the view.

- After we update the `:with_articles` trait, we can then write a new test for your fix within `spec/requests/collections_spec.rb`! 

To do this, we'll want to define a new variable, `large_collection` (or something similar), that mirrors `collection`, but that accepts our `amount` argument. We'll want to ensure that the amount we define is larger than the default amount of `3`, but not _too_ large since articles take a while to create. Something like this should work: `large_collection = create(:collection, :with_articles, amount: 6, user: user)`. Once we've defined `large_collection` and have passed it a custom amount, we'll want to write a test that ensures that the proper article count and text are returned when we visit `get "/#{user.username}/#{large_collection.articles.first.slug}"`. 

The rest of the setup for the should look fairly similar to the two other tests in the file. Here's an idea for how we could stub out the test:
```

  describe "GET collection show" do
    it "returns the proper article count and text for a large collection", :aggregate_failures do
      large_collection = create(:collection, :with_articles, amount: 6, user: user)

      get "/#{user.username}/#{large_collection.articles.first.slug}"
        ..........
    end
  end
```

Please let me know if you have any further questions or need any further guidance -- I'm happy to help! Additionally, if you feel like you'd rather have me implement the test for you, I'd be happy to do that as well. Just let me know what you're most comfortable with. 😊

File: spec/requests/collections_spec.rb [hunk 1]
```diff
@@ -17,4 +17,15 @@
       expect(response).to have_http_status(:ok)
     end
   end
+
+  describe "GET large user collection show" do
+    it "returns the proper article count and text for a large collection" do
```
Comment (juliannatetreault) [inline spec/requests/collections_spec.rb:22]: _Quick, non-blocking suggestion_:
```suggestion
    it "returns the proper article count and text for a large collection", :aggregate_failures do
```
For context, [here's](https://relishapp.com/rspec/rspec-core/docs/expectation-framework-integration/aggregating-failures) a concise explanation of what `:aggregate_failures` does!

**Review by juliannatetreault — COMMENTED** [review]
Thank you so much for taking the time to implement a test for your bug fix, @thomasstep -- you absolutely knocked it out of the park! That said, I left a quick suggestion in the spec, that once added, should make this PR good to go.

Otherwise, I tested everything out and it still works wonderfully! 😄

File: spec/requests/collections_spec.rb (hunk 1 above)
Comment (thomasstep) [reply spec/requests/collections_spec.rb]: I noticed `aggregate_failures` in your previous message, but I didn't take the time to look into what it does. Thanks for linking the documentation! 😄

Comment (juliannatetreault) [reply spec/requests/collections_spec.rb]: Absolutely! 😊

**Review by juliannatetreault — APPROVED** [review]
Thanks a million for fixing this bug and implementing a test to ensure this bug gets caught if it pops up again in the future. Excellent work, @thomasstep! 🚀

**Review by Zhao-Andy — APPROVED** [review]
This looks great! Thank you for your PR :)

**Comment by thomasstep** [comment]
@juliannatetreault Thank you so much for all your help and patience!

**Comment by juliannatetreault** [comment]
> @juliannatetreault Thank you so much for all your help and patience!

Happy to help, @thomasstep! 😄

**Pull request merged**